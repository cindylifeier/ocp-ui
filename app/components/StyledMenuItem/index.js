/**
 *
 * StyledMenuItem
 *
 */

import styled from 'styled-components';
import PropTypes from 'prop-types';
import MenuItem from 'material-ui/MenuItem';


const StyledMenuItem = styled(MenuItem)`
  display: inline-block;
`;

StyledMenuItem.propTypes = {
  /**
   * Inherited propTypes for the styled Material UI MenuItem.
   * See https://raw.githubusercontent.com/mui-org/material-ui/v0.19.4/src/MenuItem/MenuItem.js
   */
  /**
   * Location of the anchor for the popover of nested `MenuItem`
   * elements.
   * Options:
   * horizontal: [left, middle, right]
   * vertical: [top, center, bottom].
   */
  anchorOrigin: PropTypes.origin,
  /**
   * Override the default animation component used.
   */
  animation: PropTypes.func,
  /**
   * If true, a left check mark will be rendered.
   */
  checked: PropTypes.bool,
  /**
   * Elements passed as children to the underlying `ListItem`.
   */
  children: PropTypes.node,
  /**
   * @ignore
   * If true, the menu item will render with compact desktop
   * styles.
   */
  desktop: PropTypes.bool,
  /**
   * If true, the menu item will be disabled.
   */
  disabled: PropTypes.bool,
  /**
   * The focus state of the menu item. This prop is used to set the focus
   * state of the underlying `ListItem`.
   */
  focusState: PropTypes.oneOf([
    'none',
    'focused',
    'keyboard-focused',
  ]),
  /**
   * Override the inline-styles of the inner div.
   */
  innerDivStyle: PropTypes.object,
  /**
   * If true, the children will be indented.
   * This is only needed when there is no `leftIcon`.
   */
  insetChildren: PropTypes.bool,
  /**
   * The `SvgIcon` or `FontIcon` to be displayed on the left side.
   */
  leftIcon: PropTypes.element,
  /**
   * `MenuItem` elements to nest within the menu item.
   */
  menuItems: PropTypes.node,
  /**
   * Callback function fired when the menu item is touch-tapped.
   *
   * @param {object} event TouchTap event targeting the menu item.
   */
  onClick: PropTypes.func,
  /**
   * Can be used to render primary text within the menu item.
   */
  primaryText: PropTypes.node,
  /**
   * The `SvgIcon` or `FontIcon` to be displayed on the right side.
   */
  rightIcon: PropTypes.element,
  /**
   * Can be used to render secondary text within the menu item.
   */
  secondaryText: PropTypes.node,
  /**
   * Override the inline-styles of the root element.
   */
  style: PropTypes.object,
  /**
   * Location on the popover of nested `MenuItem` elements that will attach
   * to the anchor's origin.
   * Options:
   * horizontal: [left, middle, right]
   * vertical: [top, center, bottom].
   */
  targetOrigin: PropTypes.origin,
  /**
   * The value of the menu item.
   */
  value: PropTypes.any,
};

export default StyledMenuItem;
