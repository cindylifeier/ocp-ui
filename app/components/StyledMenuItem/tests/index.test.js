import React from 'react';
import { configure, mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import 'jest-styled-components';
import { MuiThemeProvider } from 'material-ui';

import StyledMenuItem from '../index';


configure({ adapter: new Adapter() });

describe('<StyledMenuItem />', () => {
  describe('snapshot tests', () => {
    it('should match snapshot', () => {
      // Arrange
      const primaryText = 'primaryText';

      // Act
      const renderedComponent = shallow(<StyledMenuItem primaryText={primaryText} />);

      // Assert
      expect(renderedComponent).toMatchSnapshot();
    });
  });

  describe('style tests', () => {
    it('should have styles', () => {
      // Arrange
      const primaryText = 'primaryText';

      // Act
      const renderedComponent = mount((
        <MuiThemeProvider>
          <StyledMenuItem primaryText={primaryText} />
        </MuiThemeProvider>));

      // Assert
      expect(renderedComponent.find(StyledMenuItem)).toHaveStyleRule('display', 'inline-block');
    });
  });
});
