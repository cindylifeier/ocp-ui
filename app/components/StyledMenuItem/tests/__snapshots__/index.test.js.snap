// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StyledMenuItem /> snapshot tests should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StyledMenuItem__StyledMenuItem
    primaryText="primaryText"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "anchorOrigin": Object {
        "horizontal": "right",
        "vertical": "top",
      },
      "checked": false,
      "className": "StyledMenuItem__StyledMenuItem-kCHNxU eynbfp",
      "desktop": false,
      "disabled": false,
      "focusState": "none",
      "insetChildren": false,
      "primaryText": "primaryText",
      "targetOrigin": Object {
        "horizontal": "left",
        "vertical": "top",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "anchorOrigin": Object {
          "horizontal": "right",
          "vertical": "top",
        },
        "checked": false,
        "className": "StyledMenuItem__StyledMenuItem-kCHNxU eynbfp",
        "desktop": false,
        "disabled": false,
        "focusState": "none",
        "insetChildren": false,
        "primaryText": "primaryText",
        "targetOrigin": Object {
          "horizontal": "left",
          "vertical": "top",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
